core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -311,4 +311,34 @@ protected static void assertValidate(QueryBuilder queryBuilder, int totalExpecte
             assertThat(queryValidationException, nullValue());
         }
     }
-}
+
+    protected String[] getRandomTypes() {
```

Comment:
did you take this code from beforeTest? If so beforeTest could call this method now?